### PR TITLE
feat: created updateGoalByID controller

### PR DESCRIPTION
**Pull request by devteni**
To test: https://goals.zuri.chat/api/v1/update/612e42849f34172f0cac4026
issue #335

### CODE REVIEW

**Comment by Asylcreek** [comment]
Please pull from the staging branch, and resolve the conflicts then make the PR again

File: routes/goalRoutes.js [hunk 1]
```diff
@@ -11,5 +11,7 @@ router.route('/').post(createGoals).get(getAllGoals);
 
 router.route('/:id').get(getSingleGoal);
 
+router.route("/:id").patch(updateGoalByID);
```
[the remark this answers is not in the document]
Comment (devteni) [reply routes/goalRoutes.js:14]: That's making a get request boss. And it's not my modification

[the remark this answers is not in the document]
Comment (Asylcreek) [reply routes/goalRoutes.js:14]: I am not asking to modify the implementation.

I am asking you to chain yours to the route.

Like so `router.route('/:id').get(getSingleGoal).patch(updateGoalByID)`

[the remark this answers is not in the document]
Comment (devteni) [reply routes/goalRoutes.js:14]: Okay, a minute

[the remark this answers is not in the document]
Comment (devteni) [reply routes/goalRoutes.js]: Done

File: routes/goalRoutes.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ router.route('/').post(createGoals).get(getAllGoals);
 
 router.route('/:id').get(getSingleGoal);
```
Comment (Asylcreek) [inline routes/goalRoutes.js:12]: You can remove this line now.

Line 14 has the implementation

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: routes/goalRoutes.js (hunk 2 above)
Comment (devteni) [reply routes/goalRoutes.js]: Done

**Comment by Asylcreek** [comment]
Just one more thing, please pull from the staging branch, resolve them conflicts then push to your repo again

**Comment by Asylcreek** [comment]
Just another thing, please pull from the staging branch, resolve them conflicts then push to your repo again

**Pull request merged**